File: src/main/kotlin/dev/murad/shipping/util/ItemHandlerVanillaContainerWrapper.java
package dev.murad.shipping.util;

import net.minecraft.world.Container;
import net.minecraft.world.item.ItemStack;
import net.neoforged.neoforge.items.ItemStackHandler;

public interface ItemHandlerVanillaContainerWrapper extends Container {

    ItemStackHandler getRawHandler();

    default int getContainerSize() {
        return getRawHandler().getSlots();
    }

    default boolean isEmpty() {
        for (int i = 0; i <  getRawHandler().getSlots(); i++) {
            if(! getRawHandler().getStackInSlot(i).isEmpty()){
                return false;
            }
        }
        return true;
    }

    default ItemStack getItem(int pIndex) {
        return getRawHandler().getStackInSlot(pIndex);
    }

    default ItemStack removeItem(int pIndex, int pCount) {
        return getRawHandler().extractItem(pIndex, pCount, false);
    }

    default ItemStack removeItemNoUpdate(int pIndex) {
        var stack = getRawHandler().getStackInSlot(pIndex);
        getRawHandler().setStackInSlot(pIndex, ItemStack.EMPTY);
        return stack;
    }

    default void setItem(int pIndex, ItemStack pStack) {
         getRawHandler().setStackInSlot(pIndex, pStack);
    }

    default void setChanged() {

    }

    default void clearContent() {
        for (int i = 0; i <  getRawHandler().getSlots(); i++) {
             getRawHandler().setStackInSlot(i, ItemStack.EMPTY);
        }
    }
}

File: src/main/kotlin/dev/murad/shipping/util/ItemHandlerVanillaContainerWrapper.kt
package dev.murad.shipping.util

import net.minecraft.world.Container
import net.minecraft.world.item.ItemStack
import net.neoforged.neoforge.items.ItemStackHandler

interface ItemHandlerVanillaContainerWrapper : Container {
    
    fun getRawHandler(): ItemStackHandler

    override fun getContainerSize(): Int {
        return getRawHandler().slots
    }

    override fun isEmpty(): Boolean {
        for (i in 0 until getRawHandler().slots) {
            if (!getRawHandler().getStackInSlot(i).isEmpty) {
                return false
            }
        }
        return true
    }

    override fun getItem(pIndex: Int): ItemStack {
        return getRawHandler().getStackInSlot(pIndex)
    }

    override fun removeItem(pIndex: Int, pCount: Int): ItemStack {
        return getRawHandler().extractItem(pIndex, pCount, false)
    }

    override fun removeItemNoUpdate(pIndex: Int): ItemStack {
        val stack = getRawHandler().getStackInSlot(pIndex)
        getRawHandler().setStackInSlot(pIndex, ItemStack.EMPTY)
        return stack
    }

    override fun setItem(pIndex: Int, pStack: ItemStack) {
        getRawHandler().setStackInSlot(pIndex, pStack)
    }

    override fun setChanged() {
    }

    override fun clearContent() {
        for (i in 0 until getRawHandler().slots) {
            getRawHandler().setStackInSlot(i, ItemStack.EMPTY)
        }
    }
}
